keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -149,22 +149,17 @@ metadata:
   finalizers:
     - cosi.io/finalizer [2]
 spec:
-  protocol:
-    name: [3]
-    version: [4]
-  bucketPrefix: [5]
-  bucketClassName: [6]
-  bucketInstanceName: [7]
+  bucketPrefix: [3]
+  bucketClassName: [4]
+  bucketInstanceName: [5]
 status:
-  bucketAvailable: [8]
+  bucketAvailable: [6]
 ```
 
 1. `labels`: added by COSI.  Key’s value should be the provisioner name. Characters that do not adhere to [Kubernetes label conventions](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) will be converted to ‘-’.
 1. `finalizers`: added by COSI to defer `BucketRequest` deletion until backend deletion succeeds.
-1. `protocol.name`: (required) specifies the desired protocol.  One of {“s3”, “gs”, or “azureBlob”}.
-1. `protocol.version`: (optional) specifies the desired version of the `protocol`. For "s3", a value of "v2" or "v4" could be used. 
 1. `bucketPrefix`: (optional for greenfield, ignored for brownfield) prefix prepended to a generated new bucket name, eg. “yosemite-photos-". If `bucketInstanceName` is supplied then `bucketPrefix` is ignored because the request is for access to an existing bucket.
```

Comment:
Is it fair to say that `bucketPrefix` and `bucketInstanceName` are mutually exclusive?  Can we imagine a 3rd option that might expat that into a more oneof-like pattern?  Or are we REALLY REALLY confident it will only be 2?

Reply:
Today, a user setting `bucketInstanceName` is a brownfield-only use case. Since the underlying bucket exists, and the `Bucket` abstraction instance exists, this field is the link. I don't see how a bucket prefix can be used in brownfield unless, down the road we incorporate that prefix in name(s) of supporting COSI resources, eg `Bucket.Name`, etc.

For greenfield, there should be no existing `Bucket` instance, and no existing backend bucket, and thus no need to use BR.bucketInstance. I guess it's conceivable that a pre-existing `Bucket` instance could be referenced in greenfield to bypass or override the BC?

Reply:
so "yes, they are mutually exclusive" ?

Reply:
yes, in the kep, they are mutually exclusive. Should that be written more clearly?

Reply:
Clarity is always good, this feeds into my larger concerns

Reply:
add mutual exclusion suggestion to kep.